src/query-client/hooks/role.js

```diff
@@ -0,0 +1,54 @@
+import { useQuery } from 'react-query';
+import { List, Map } from 'immutable';
+import {
+  buildMemberRoleKey,
+  buildRoleKey,
+  OWN_ROLES_KEY,
+} from '../config/keys';
+import * as Api from '../api';
+
+export default (queryClient, queryConfig) => {
+  const { retry, cacheTime, staleTime } = queryConfig;
+  const defaultOptions = {
+    retry,
+    cacheTime,
+    staleTime,
+  };
+
+  const useOwnRoles = () =>
+    useQuery({
+      queryKey: OWN_ROLES_KEY,
+      queryFn: () => Api.getOwnRoles(queryConfig).then((data) => List(data)),
+      onSuccess: async (roles) => {
+        // save items in their own key
+        // eslint-disable-next-line no-unused-expressions
+        roles?.forEach(async (role) => {
```

Comment:
Does this mean you can receive `undefined`? Maybe you should receive an empty array.